core/templates/services/playthrough.service.ts

```diff
@@ -45,7 +45,7 @@ class CyclicStateTransitionsTracker {
 
   constructor(initStateName: string) {
     this.pathOfVisitedStates = [initStateName];
-    this.cycleOfVisitedStates = null;
+    this.cycleOfVisitedStates = [];
```

Comment:
Why is this changed? 

(Note that any changes to the code itself need to make sense conceptually, not just "because it doesn't follow the type". I.e. it would be good to have an explanation of why the code as it stands is currently wrong, rather than the typeinfo needing to be changed. You can provide an explanation via replying to these review comments, or by leaving comments in a self-review.)

Reply:
Previously i have added `' '` but after reviewing it with vojtechjelinek , i did this as i cant add anything in it due to being string type since i have to initialise it with something.